src/main/java/hudson/scm/SubversionSCM.java

```diff
@@ -752,9 +752,11 @@ private void calcChangeLog(Run<?,?> build, FilePath workspace, File changelogFil
         // maybe some XSLT engine doesn't close the stream properly.
         // so let's do it by ourselves to be really sure that the stream gets closed.
         OutputStream os = new BufferedOutputStream(new FileOutputStream(changelogFile));
-        boolean created;
+        boolean created = false;
         try {
-            created = new SubversionChangeLogBuilder(build, workspace, (SVNRevisionState) baseline, env, listener, this).run(externalsMap, new StreamResult(os));
+            if(baseline instanceof SVNRevisionState){
+                created = new SubversionChangeLogBuilder(build, workspace, (SVNRevisionState) baseline, env, listener, this).run(externalsMap, new StreamResult(os));
+            }
```

Comment:
else print warning (e.g. to the system log)?

Reply:
I think, logging will be useful only for debugin purpoises. I will make it in my own environment to find the source of ClassCastException problem.
If I find (soon or later I cannot predict from this point) I will make new PR.
Now, my current PR will save many nerves for users of subversion-plugin :) Hope, you agree with me

Reply:
I agree for sure, it is not a blocker from my PoV. I just approved it with a suggestion

Reply:
Thank you, Oleg!